Source/WebCore/animation/ViewTimeline.cpp

```diff
@@ -138,4 +142,75 @@ AnimationTimeline::ShouldUpdateAnimationsAndSendEvents ViewTimeline::documentWil
     return AnimationTimeline::ShouldUpdateAnimationsAndSendEvents::No;
 }
 
+Element* ViewTimeline::source() const
+{
+    if (auto* sourceRender = sourceRenderer())
+        return sourceRender->element();
+    return nullptr;
+}
+
+RenderBox* ViewTimeline::sourceRenderer() const
+{
+    if (!m_subject)
+        return nullptr;
+
+    Ref document = m_subject->document();
+
+    CheckedPtr subjectRenderer = m_subject->renderer();
```

Comment:
This makes me think we need some WPT that try to use scroll timeline on something with `display:contents` (which means the element won't have a renderer, but its children will).

Reply:
Added this to the list in the quip.